### PR TITLE
test: update expected for Rdatasets

### PR DESCRIPTION
**Pull request by bkmgit**
Address https://github.com/red-data-tools/red-datasets/issues/111

### CODE REVIEW

**Comment by bkmgit** [comment]
Next problem is that https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/multilabel.html is not valid xml, the section `'>', '<',` should be `'&gt', '&lt',`

**Comment by bkmgit** [comment]
Tests now pass. Maybe there is a more elegant way to fix the XML errors in https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/multilabel.html ? Alternatively, maybe the datasets which are not too large could be archived in one place specific for red-datasets and a description of their origin also indicated. This would ensure more stable operation.

**Comment by kou** [comment]
> Next problem is that https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/multilabel.html is not valid xml, the section `'>', '<',` should be `'&gt', '&lt',`

Could you report this to libsvm?

**Comment by bkmgit** [comment]
> Could you report this to LIBSVM?

Done

File: lib/datasets/libsvm-dataset-list.rb [hunk 1]
```diff
@@ -78,6 +78,13 @@ def extract_text(element)
     def open_detail(detail)
       data_path = cache_dir_path + detail
       download(data_path, @metadata.url + detail)
+      # Hack to fix XML error in EUR-Lex section of
+      # https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/multilabel.html
+      if detail == "multilabel.html"
+        page = data_path.read
+        page.gsub!("'>', '<',", "'&gt', '&lt',")
+        data_path.write(page)
```
Comment (kou) [inline lib/datasets/libsvm-dataset-list.rb:86]: The downloaded file is cached.
So we should not write substituted content.

Could you substitute the content in memory?

```ruby
data_path.open do |input|
  if detail == "multilabel.html"
    yield(input.gsub("'>', '<',", "'&gt', '&lt',"))
  else
    yield(input)
  end
end
```

Comment (bkmgit) [reply lib/datasets/libsvm-dataset-list.rb:86]: The site has been updated. Maybe this change can be reverted? Can also look at later updating the code to parse HTML.

Comment (bkmgit) [reply lib/datasets/libsvm-dataset-list.rb]: While cleaner input.gsub fails with 
```
Error: test: multi-label(LIBSVMDatasetTest): NoMethodError: undefined method `gsub' for #<File:0x00007f90436c0bf8>
```
input is a file object, so may need some other way to modify it. As the site has been updated, can revert to original code.

Comment (kou) [reply lib/datasets/libsvm-dataset-list.rb]: Ah, sorry. `.read` was missed: `input.read.gsub(...)`

**Comment by kou** [comment]
Thanks.

**Pull request merged**